src/sagemaker_xgboost_container/data_utils.py

```diff
@@ -622,3 +622,39 @@ def _make_symlink(path, source_path, name, index):
     base_name = os.path.join(source_path, name + '_' + str(index))
     logging.info('creating symlink between Path {} and destination {}'.format(source_path, base_name))
     os.symlink(path, base_name)
+
+
+def check_data_redundancy(train_path, validate_path):
+    """Give a hint about whether same files exist in the train/validation folder.
+
+    The validation score of models would make less sense if there are same data in both training/validation folder.
+    This function is used to give warnings when there are files sharing the same name and file size in training and
+    validation folders
+
+    param train_path : path to training data
+    param validate_path : path to validation data
+
+    """
+    if not os.path.exists(train_path):
+        exc.UserError("train path is not existed")
+        if not os.path.exists(validate_path):
+            exc.UserError("validate path is not existed")
+
+    isRedundant = False
+    training_files_set = set(f for f in os.listdir(train_path) if os.path.isfile(os.path.join(train_path, f)))
+    validation_files_set = set(f for f in os.listdir(validate_path) if os.path.isfile(os.path.join(validate_path, f)))
+    same_name_files = training_files_set.intersection(validation_files_set)
+    if len(same_name_files) != 0:
+        for f in same_name_files:
+            f_train_name = os.path.join(train_path, f)
```

Comment:
nit: f_train_name is really f_train_path or like full_train_path, ditto for f_validate_name.

Also really f should be fname as its really a string (a name) and not a file object but I think that's fine here for concision.